components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1464,53 +1463,49 @@ void RewardsServiceImpl::SetConfirmationsIsReady(const bool is_ready) {
     ads_service->SetConfirmationsIsReady(is_ready);
 }
 
-void RewardsServiceImpl::GetAdsNotificationsHistory() {
+void RewardsServiceImpl::ConfirmationsTransactionHistoryDidChange() {
+    for (auto& observer : observers_)
+    observer.OnConfirmationsHistoryChanged(this);
+}
+
+void RewardsServiceImpl::GetAdsNotificationsHistory(
+    brave_rewards::AdsNotificationsHistoryCallback callback) {
   if (!Connected()) {
     return;
   }
 
   auto earnings_range = GetEarningsRange();
 
-  bat_ledger_->GetAdsNotificationsHistory(earnings_range.first,
-      earnings_range.second, base::BindOnce(
-      &RewardsServiceImpl::OnGetAdsNotificationsHistoryMojoProxy, AsWeakPtr()));
-}
-
-void RewardsServiceImpl::ConfirmationsTransactionHistoryDidChange() {
+  bat_ledger_->GetAdsNotificationsHistory(
+      earnings_range.first,
+      earnings_range.second,
+      base::BindOnce(&RewardsServiceImpl::OnGetAdsNotificationsHistoryMojoProxy,
+                     AsWeakPtr(),
+                     std::move(callback)));
 }
 
 void RewardsServiceImpl::OnGetAdsNotificationsHistoryMojoProxy(
+    brave_rewards::AdsNotificationsHistoryCallback callback,
     const std::string& transactions) {
   std::unique_ptr<ledger::TransactionsInfo> info;
   if (!transactions.empty()) {
     info.reset(new ledger::TransactionsInfo());
     info->FromJson(transactions);
   }
-  OnGetAdsNotificationsHistory(std::move(info));
-}
 
-void RewardsServiceImpl::OnGetAdsNotificationsHistory(
-    std::unique_ptr<ledger::TransactionsInfo> transactions_info) {
-  if (!transactions_info) {
+  if (!info) {
```

Comment:
We can't just not respond to the callback. It should always gets called and probably just return 0,0 in this case?

Reply:
done